text/0000-buildpack-input-vars.md

```diff
@@ -0,0 +1,142 @@
+# Meta
+[meta]: #meta
+- Name: Replace positional args to Buildpack executables with env vars
+- Start Date: 2021-11-18
+- Author(s): [jkutner](https://github.com/jkutner)
+- Status: Draft Draft
+- RFC Pull Request: (leave blank)
+- CNB Pull Request: (leave blank)
+- CNB Issue: (leave blank)
+- Supersedes: N/A
+
+# Summary
+[summary]: #summary
+
+This is a proposal to replace the positional arguments of Buildpack executables with named environment variables.
+
+# Definitions
+[definitions]: #definitions
+
+- *positional arguments* - values passed into an executable command such that they are accessible as `$1`, `$2`, etc
+
+# Motivation
+[motivation]: #motivation
+
+Postional arguments to Buildpack executables have been the way for the buildpack execution engine to provide inputs for buildpacks since buildpacks v1 was created. However, positional arguments are limiting, and what they represent is not obvious without reading the spec.
+
+In the spec today they are defined as:
+
+* `/bin/detect <platform[AR]> <plan[E]>`
+* `/bin/build <layers[EIC]> <platform[AR]> <plan[ER]>`
+
+Using env vars helps make these inputs more easily accesible from language bindings like [libcnb.bash](https://github.com/jkutner/libcnb.bash).
+
+# What it is
+[what-it-is]: #what-it-is
+
+We will deprecate the positional arguments to `bin/detect` and `bin/build` with the following environment variables:
+
+### bin/detect
+
+* `CNB_PLATFORM_DIR` - replaces the first positional argument to `bin/build`. Uses the same env var name as the Platform spec.
+* `CNB_PLAN_PATH` - replaces the second positional argument to `bin/build`. Uses the same env var name as the Platform spec.
```

Comment:
as far as naming goes, do we use `DIR` for directories and `PATH` for files? Just wondering if we should take the time to standardize here (if needed).

Reply:
I believe @sclevine was pretty consistent with this in other places in the spec, which is why I adopted it